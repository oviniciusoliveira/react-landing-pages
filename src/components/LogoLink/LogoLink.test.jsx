import { renderTheme } from '../../test/render-theme';
import { LogoLink } from '.';
import { screen } from '@testing-library/dom';

describe('<LogoLink />', () => {
  test('should render text logo with link', () => {
    renderTheme(<LogoLink text="Logo" link="#target" />);

    const heading = screen.getByRole('heading', { name: 'Logo' });
    expect(heading).toBeInTheDocument();

    const link = screen.getByRole('link', { name: 'Logo' });
    expect(link).toHaveAttribute('href', '#target');
  });

  test('should render image logo with link', () => {
    renderTheme(<LogoLink text="Logo" link="#target" imageSrc="image.jpg" />);

    const link = screen.getByRole('link', { name: 'Logo' });
    expect(link).toHaveAttribute('href', '#target');

    const image = screen.getByAltText('Logo');
    expect(image).toHaveAttribute('src', 'image.jpg');
  });

  test('should match snapshot', () => {
    const { container } = renderTheme(<LogoLink text="Logo" link="#target" imageSrc="image.jpg" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
