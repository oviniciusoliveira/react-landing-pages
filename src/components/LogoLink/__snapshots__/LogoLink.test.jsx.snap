// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > img {
  height: 3rem;
}

.c0 {
  color: #0A1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="Logo"
      src="image.jpg"
    />
  </a>
</h1>
`;
